staging/src/k8s.io/externaljwt/pkg/plugin/plugin.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"gopkg.in/square/go-jose.v2/jwt"
+	"k8s.io/externaljwt"
+	externaljwtv1alpha1 "k8s.io/externaljwt/apis/v1alpha1"
+	externaljwtmetrics "k8s.io/externaljwt/pkg/metrics"
+)
+
+func init() {
+	externaljwtmetrics.RegisterMetrics()
+}
+
+type VerificationKeys struct {
+	Keys          []externaljwt.PublicKey
+	DataTimestamp time.Time
+}
+
+// New calls external signer to fill out supported keys.
+// It also starts a periodic sync of external keys.
+// In order for the key cache and external signing to work correctly, pass a context that will live as
+// long as the dependent process; is used to maintain the lifetime of the connection to external signer.
+func New(ctx context.Context, issuer, socketPath string, keySyncTimeout time.Duration, preventSigningWithNonOIDCKeys bool) (*Plugin, *keyCache, error) {
+	conn, err := grpc.Dial(
+		socketPath,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithAuthority("localhost"),
+		grpc.WithDefaultCallOptions(grpc.WaitForReady(true)),
+		grpc.WithContextDialer(func(ctx context.Context, path string) (net.Conn, error) {
+			return (&net.Dialer{}).DialContext(ctx, "unix", path)
+		}),
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf("while dialing grpc socket at %q: %w", socketPath, err)
+	}
+
+	plugin := newPlugin(issuer, conn, preventSigningWithNonOIDCKeys)
+
+	initialFillCtx, cancel := context.WithTimeout(ctx, keySyncTimeout)
+	defer cancel()
+
+	if err := plugin.keyCache.initialFill(initialFillCtx); err != nil {
+		return nil, nil, fmt.Errorf("while initially filling key cache: %w", err)
+	}
+	go plugin.keyCache.scheduleSync(ctx, keySyncTimeout)
+
+	go func() {
+		<-ctx.Done()
+		conn.Close()
+	}()
+
+	return plugin, plugin.keyCache, nil
+}
+
+// enables plugging in an external jwt signer.
+type Plugin struct {
+	iss                           string
+	client                        externaljwtv1alpha1.ExternalJWTSignerClient
+	keyCache                      *keyCache
+	preventSigningWithNonOIDCKeys bool
```

Comment:
Flip this `bool` so a caller passing in `false` does the right thing:

```suggestion
	allowSigningWithNonOIDCKeys bool
```

Reply:
Done